subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c

```diff
@@ -0,0 +1,603 @@
+	case RP_CC_EVT_CIS_REQ_REJECT:
+		/* Continue procedure in next prepare run */
+		ctx->state = RP_CC_STATE_WAIT_TX_REJECT_IND;
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+}
+
+
+static void rp_cc_state_wait_instant(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				     void *param)
+{
+	switch (evt) {
+	case RP_CC_EVT_RUN:
+		rp_cc_check_instant(conn, ctx, evt, param);
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+}
+
+
+static void rp_cc_state_wait_cis_established(struct ll_conn *conn, struct proc_ctx *ctx,
+					     uint8_t evt, void *param)
+{
+	struct pdu_data *pdu = (struct pdu_data *)param;
+
+	switch (evt) {
+	case RP_CC_EVT_RUN:
+		/* Check for CIS state */
+		if (rp_cc_check_cis_established_lll(conn, ctx, pdu)) {
+			/* CIS was established, so let's got ahead and complete procedure */
+			rp_cc_complete(conn, ctx, evt, param);
+		}
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+
+}
+
+
+static void rp_cc_execute_fsm(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	switch (ctx->state) {
+	/* Create Procedure */
+	case RP_CC_STATE_IDLE:
+		rp_cc_state_idle(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_RX_CIS_REQ:
+		rp_cc_state_wait_rx_cis_req(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_NTF_CIS_CREATE:
+		rp_cc_state_wait_ntf_cis_create(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_TX_REJECT_IND:
+		rp_cc_state_wait_tx_reject_ind(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_TX_CIS_RSP:
+		rp_cc_state_wait_tx_cis_rsp(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_REPLY:
+		rp_cc_state_wait_reply(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_RX_CIS_IND:
+		rp_cc_state_wait_rx_cis_ind(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_INSTANT:
+		rp_cc_state_wait_instant(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_CIS_ESTABLISHED:
+		rp_cc_state_wait_cis_established(conn, ctx, evt, param);
+		break;
+	case RP_CC_STATE_WAIT_NTF:
+		rp_cc_state_wait_ntf(conn, ctx, evt, param);
+		break;
+	default:
+		/* Unknown state */
+		LL_ASSERT(0);
+	}
+}
+
+void llcp_rp_cc_rx(struct ll_conn *conn, struct proc_ctx *ctx, struct node_rx_pdu *rx)
+{
+	struct pdu_data *pdu = (struct pdu_data *)rx->pdu;
+
+	switch (pdu->llctrl.opcode) {
+	case PDU_DATA_LLCTRL_TYPE_CIS_REQ:
+		rp_cc_execute_fsm(conn, ctx, RP_CC_EVT_CIS_REQ, pdu);
+		break;
+	case PDU_DATA_LLCTRL_TYPE_CIS_IND:
+		rp_cc_execute_fsm(conn, ctx, RP_CC_EVT_CIS_IND, pdu);
+		break;
+	case PDU_DATA_LLCTRL_TYPE_REJECT_IND:
+	case PDU_DATA_LLCTRL_TYPE_REJECT_EXT_IND:
+		rp_cc_execute_fsm(conn, ctx, RP_CC_EVT_REJECT, pdu);
+		break;
+	default:
+		/* Unknown opcode */
+		LL_ASSERT(0);
+	}
+}
+
+void llcp_rp_cc_init_proc(struct proc_ctx *ctx)
+{
+	switch (ctx->proc) {
+	case PROC_CIS_CREATE:
+		ctx->state = RP_CC_STATE_IDLE;
+		break;
+	default:
+		LL_ASSERT(0);
+	}
+}
+
+bool llcp_rp_cc_awaiting_reply(struct proc_ctx *ctx)
```

Comment:
```suggestion
bool llcp_rp_cc_awaiting_reply(const struct proc_ctx *ctx)
```

Reply:
Skipping this, as I will not do this change everywhere just now;-)